### PR TITLE
Make response headers for HEAD requests match those for GET requests

### PR DESCRIPTION
**Pull request by harringj**
This PR makes HEAD response headers match the non-payload headers returned by a GET request to the same resource. Specifically, this PR ensures the "Vary:" and "Preference-Applied” headers are present for HEAD.

Resolves: https://jira.duraspace.org/browse/FCREPO-2661

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -205,15 +205,20 @@ protected Response getContent(final String rangeValue,
...
+        setVaryAndPreferenceAppliedHeaders(servletResponse, prefer);
         return ok(outputStream).build();
     }
 
+    protected void setVaryAndPreferenceAppliedHeaders(final HttpServletResponse servletResponse,
+            final MultiPrefer prefer) {
+        if (prefer != null) {
+            prefer.getReturn().addResponseHeaders(servletResponse);
+        }
+    servletResponse.addHeader("Vary", "Accept, Range, Accept-Encoding, Accept-Language");
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: formatting looks shifted.

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 2]
```diff
@@ -281,13 +281,14 @@ private void testHeadDefaultContentType(final String mimeType) throws IOExceptio
...
     }
-    
+
     private void testHeadVaryAndPreferHeaders(final CloseableHttpResponse response) {
-    	final Collection<String> preferenceApplied = getHeader(response, "Preference-Applied");
-    	final Collection<String> vary = getHeader(response, "Vary");
-    	assertTrue("Didn't find valid Preference-Applied header", preferenceApplied.contains("return=representation"));
-    	assertTrue("Didn't find valid Vary Prefer header", vary.contains("Prefer"));
-    	assertTrue("Didn't find valid Vary Accept headers", vary.contains("Accept, Range, Accept-Encoding, Accept-Language"));
+        final Collection<String> preferenceApplied = getHeader(response, "Preference-Applied");
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java]: It is difficult to tell if there are functional changes here, or just reformatting.
Could you please revert the formatting changes to make the functional changes more clear?

**Comment by awoods** [comment]
In testing this, I am seeing different behavior for Binary resources.
Example, create a resource: `http://localhost:8080/rest/image`
```bash
$curl http://localhost:8080/rest/image -H"Prefer: return=minimal" -I
```
```
HTTP/1.1 200 OK
Date: Tue, 06 Mar 2018 22:05:46 GMT
ETag: "ba89f667e3bee15a2aca6ffd7569d4d95738c896"
Last-Modified: Tue, 06 Mar 2018 22:05:38 GMT
Content-Type: image/png
Accept-Ranges: bytes
Content-Disposition: attachment; filename="Screenshot from 2017-06-21 19-17-54.png"; creation-date="Tue, 06 Mar 2018 22:05:38 GMT"; modification-date="Tue, 06 Mar 2018 22:05:38 GMT"; size=120656
Link: <http://www.w3.org/ns/ldp#Resource>;rel="type"
Link: <http://www.w3.org/ns/ldp#NonRDFSource>;rel="type"
Link: <http://localhost:8080/rest/image/fcr:metadata>; rel="describedby"
Allow: DELETE,HEAD,GET,PUT,OPTIONS
Preference-Applied: return=minimal
Vary: Prefer
Vary: Accept, Range, Accept-Encoding, Accept-Language
Content-Length: 120656
Server: Jetty(9.2.3.v20140905)
```
```bash
$ curl http://localhost:8080/rest/image -H"Prefer: return=minimal" -i
HTTP/1.1 200 OK
Date: Tue, 06 Mar 2018 22:05:54 GMT
ETag: "ba89f667e3bee15a2aca6ffd7569d4d95738c896"
Last-Modified: Tue, 06 Mar 2018 22:05:38 GMT
Content-Type: image/png
Accept-Ranges: bytes
Content-Disposition: attachment; filename="Screenshot from 2017-06-21 19-17-54.png"; creation-date="Tue, 06 Mar 2018 22:05:38 GMT"; modification-date="Tue, 06 Mar 2018 22:05:38 GMT"; size=120656
Link: <http://www.w3.org/ns/ldp#Resource>;rel="type"
Link: <http://www.w3.org/ns/ldp#NonRDFSource>;rel="type"
Link: <http://localhost:8080/rest/image/fcr:metadata>; rel="describedby"
Allow: DELETE,HEAD,GET,PUT,OPTIONS
ETag: "ba89f667e3bee15a2aca6ffd7569d4d95738c896"
Last-Modified: Tue, 06 Mar 2018 22:05:38 GMT
Cache-Control: no-transform, must-revalidate, max-age=0
Content-Length: 120656
Server: Jetty(9.2.3.v20140905)
```
Notice the presence of `Cache-Control`, and the duplicated `ETag` and `Last-Modified` headers.

**Comment by harringj** [comment]
Closing this PR and opening a new one.

**Pull request closed without merging**